mlx_lm/tuner/callbacks.py

```diff
@@ -31,13 +31,19 @@ def __init__(
             )
         self.wrapped_callback = wrapped_callback
         wandb.init(project=project_name, dir=log_dir, config=config)
+    
+    def _convert_to_serializable(self, data: dict) -> dict:
+        return {
+            k: (v.item() if hasattr(v, "item") else v.tolist() if hasattr(v, "tolist") else v)
```

Comment:
Part of the problem here is we don't know what arrays are scalars and what arrays need to be logged as lists. It is an error to call `a.item` if `a` is not a single value but it will still have the attribute.

If you want arrays logged as scalars I think we should require that one call item ahead of time. And then standardize using `tolist` on all `mx.array` which are input to the logger directly. Does that make sense?